### PR TITLE
HV: fix expression is not boolean

### PR DESCRIPTION
**Pull request by HuiHuangShi**
MISRA-C standard requires the type of result of expression in if/while pattern shall be boolean.

Tracked-On: #861
Signed-off-by: Huihuang Shi <huihuang.shi@intel.com>

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: hypervisor/boot/cmdline.c [hunk 1]
```diff
@@ -20,14 +20,14 @@ int32_t parse_hv_cmdline(void)
 	struct multiboot_info *mbi = NULL;
 
 	if (boot_regs[0] != MULTIBOOT_INFO_MAGIC) {
-		ASSERT(0, "no multiboot info found");
+		ASSERT(false, "no multiboot info found");
```
Comment (dongyaozu) [inline hypervisor/boot/cmdline.c]: this is useless then...  Remove?

**Comment by HuiHuangShi** [comment]
out of time

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Pull request merged**